libnavigation-core/src/test/java/com/mapbox/navigation/core/NavigationSessionTest.kt

```diff
@@ -34,7 +35,7 @@ class NavigationSessionTest {
 
     @Test
     fun drive_only() {
-        navigationSession.onSessionStarted()
+        navigationSession.onSessionStateChanged(TripSessionState.STARTED)
```

Comment:
NIT Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert
If the `A`rrange part gets too long we should create factory or builder methods.

The same thing applies to the rest of the tests 👇

Reply:
If I understand you correctly, there's no arrangement here. We are running one scenario with no additional setup and asserting the results.

Reply:
It's "comestics" 😅 

```kotlin
@Test
fun drive_only() {
    // Arrange
    
    // Act
    
    // Assert
}
```

Note the empty lines between Arrange / Act and Act / Assert.

Also I'm not a big fan of inline setup in tests because among other things it hides the Arrange part completely.